packages/venia-concept/src/util/makeUrl.js

```diff
@@ -39,36 +38,42 @@ const mediaBases = new Map()
  * @param {number} props.width - the desired resize width of the image
  */
 const makeOptimizedUrl = (path, { type, width } = {}) => {
-    const isAbsolute = absoluteUrlRE.test(path);
-    const shouldOptimizeByDefault =
-        !isAbsolute && type && type.startsWith('image-');
-    const fastlyBase = shouldOptimizeByDefault && fastlyOrigin;
-    const baseUrl = fastlyBase || window.location.href;
-
-    const urlObject = new URL(path, baseUrl);
-    const params = new URLSearchParams(urlObject.search);
+    const { href, origin } = window.location;
+    let urlObject = new URL(path, href);
 
-    if (!isAbsolute && type && mediaBases.has(type)) {
+    if (type && mediaBases.has(type)) {
         const mediaBase = mediaBases.get(type);
         // prepend media base if it isn't already part of the pathname
         if (!urlObject.pathname.includes(mediaBase)) {
             urlObject.pathname = mediaBase + urlObject.pathname;
         }
+        // add image optimization parameters and optionally change origin
+        if (type.startsWith('image-')) {
```

Comment:
With this nested inside of `if (type && mediaBases.has(type))` we're back to only processing types in the Map. This gnarly nested if should bring that functionality back:

```js
if (type) {
    if (mediaBases.has(type)) {
        const mediaBase = mediaBases.get(type);
        if (!urlObject.pathname.includes(mediaBase)) {
            urlObject.pathname = mediaBase + urlObject.pathname;
        }
    }
    ...
}
```

Reply:
Very good catch, thank you. Switched.